src/halerror.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+angular.module('hypermedia')
+
+  .run(function ($q, ResourceContext, HalError) {
+    var vndErrorHandler = function (response) {
+      response.error = new HalError(response.data);
+
+      return $q.reject(response);
+    };
+
+    ResourceContext.registerErrorHandler('application/vnd+error', vndErrorHandler);
+  })
+
+  .factory('HalError', function () {
```

Comment:
Shall we call it something like `VndError`? (I feel it is not HAL, just compatible with it.)